### PR TITLE
Feat ppu2

### PR DESCRIPTION
**Pull request by jtwhite79**
Initial support to automatically use ppu for kriging solve if available @briochh it's a hack- open to suggestions

### CODE REVIEW

File: etc/environment.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ channels:
 dependencies:
   # required
   - python>=3.8
-  - numpy
+  - numpy<2.0
```
Comment (briochh) [inline etc/environment.yml:7]: I think, we are supporting numpy 2 now... or have you detected issues?

File: pyemu/utils/geostats.py [hunk 2]
```diff
@@ -872,6 +873,8 @@ def calc_factors_grid(
         var_filename=None,
         forgive=False,
         num_threads=1,
+        try_use_ppu=False,
```
Comment (briochh) [inline pyemu/utils/geostats.py:876]: let's aim to phase this the default to True, over time.
Any danger of some docstring additions though?

File: pyemu/utils/geostats.py [hunk 3]
```diff
@@ -940,6 +943,58 @@ def calc_factors_grid(
...
+                                                          self.point_data.y.values,zns,
+                                                          x.ravel(),y.ravel(),znt,
+                                                          vartype,krigtype,
+                                                          v.a,v.anisotropy,v.bearing,
+                                                          search_radius,maxpts_interp,
+                                                          minpts_interp,ppu_factor_filename,
+                                                          factorfiletype)
+            plib.free_all_memory()
+            assert os.path.exists(ppu_factor_filename)
+            reformat_factorfile(x.shape[0], x.shape[1], self.point_data, self.geostruct, ppu_factor_filename)
```
Comment (briochh) [inline pyemu/utils/geostats.py:992]: is the reformatting for downstream compatibility? with our current pyemu methods? pestutils? is there anything we can do about that?

File: etc/environment.yml (hunk 1 above)
Comment (jtwhite79) [reply etc/environment.yml:7]: No issues - this was a legacy thing that snuck in.  We are good with np > 2 yea?

File: pyemu/utils/geostats.py (hunk 2 above)
Comment (jtwhite79) [reply pyemu/utils/geostats.py:876]: Good points.  I was thinking of a "soft launch" here to ease into it but maybe a docstring would be less ambiguous

File: pyemu/utils/geostats.py (hunk 3 above)
Comment (jtwhite79) [reply pyemu/utils/geostats.py:992]: I tried to find a way around this but the new fac file format coming out of ppu doesn't have array shape info in the header so I thought it was easiest to patch that here when we know the shape.

File: etc/environment.yml (hunk 1 above)
Comment (briochh) [reply etc/environment.yml:7]: I "_think_" so

**Comment by jtwhite79** [comment]
@briochh the one thing this PR doesn't do (yet) is weave into `PstFrom.add_parameters()`.  I think it could be as simple as passing an explicit `try_use_ppu=True` where we call `calc_factors_grid()` - probably would be wise to make a new optional arg for `add_parameters()` until we get some more in-the-field testing and experience with it (maybe call it `try_use_ppu` also?).  That sound ok with you?

**Pull request closed without merging**